nni/algorithms/compression/pytorch/auto_compress/auto_compress_engine.py

```diff
@@ -0,0 +1,150 @@
+            finetune_trainer(model)
+        return model
+
+    @classmethod
+    def __compress_quantization_pipeline(cls, algorithm_name: str,
+                                         model: Module,
+                                         config_list: list,
+                                         evaluator: Callable[[Module], float],
+                                         optimizer: Optional[Optimizer],
+                                         trainer: Callable[[Module, Optimizer], None],
+                                         finetune_trainer: Optional[Callable[[Module, Optimizer], None]],
+                                         **compressor_parameter_dict) -> Module:
+        # evaluator is for future use
+        quantizer = QUANTIZER_DICT[algorithm_name](model, config_list, optimizer, **compressor_parameter_dict)
+        model = quantizer.compress()
+        if trainer:
+            trainer(model)
+        if finetune_trainer:
+            finetune_trainer(model)
+        return model
+
+    @classmethod
+    def _compress_pipeline(cls, compressor_type: str,
+                           algorithm_name: str,
+                           model: Module,
+                           config_list: list,
+                           evaluator: Callable[[Module], float],
+                           optimizer: Optional[Optimizer],
+                           trainer: Optional[Callable[[Module, Optimizer], None]],
+                           finetune_trainer: Optional[Callable[[Module, Optimizer], None]],
+                           **compressor_parameter_dict) -> Module:
+        func_dict = {
+            'pruner': cls.__compress_pruning_pipeline,
+            'quantizer': cls.__compress_quantization_pipeline
+        }
+        _logger.info('%s compressor config_list:\n%s', algorithm_name, json_tricks.dumps(config_list, indent=4))
+        return func_dict[compressor_type](algorithm_name, model, config_list, evaluator, optimizer, trainer,
+                                          finetune_trainer, **compressor_parameter_dict)
+
+    @classmethod
+    def trial_execute_compress(cls):
+        auto_compress_module = import_('auto_compress_module.AutoCompressModule')
+
+        parameter = nni.get_next_parameter()['compressor_type']
+        compressor_type, algorithm_config = parameter['_name'], parameter['algorithm_name']
+        algorithm_name = algorithm_config['_name']
+        converted_config_dict = {k: v for k, v in algorithm_config.items() if k.startswith('config_list::')}
+        parameter_dict = {k.split('parameter::')[1]: v for k, v in algorithm_config.items() if k.startswith('parameter::')}
+
+        config_list = cls._convert_config_list(compressor_type, converted_config_dict)
+
+        model, evaluator, optimizer = auto_compress_module.model(), auto_compress_module.evaluator(), auto_compress_module.optimizer()
+        trainer = auto_compress_module.trainer(compressor_type, algorithm_name)
```

Comment:
Just curious. I haven't found the implementation of `trainer` in example. So in the shown case, we only finetune model without training it during compress pipeline? Or they represent the same thing so we can ignore 'trainer'.